src/DurableTask.Netherite/Scaling/AzureBlobLoadPublisher.cs

```diff
@@ -50,44 +50,96 @@ Task UploadPartitionInfo(uint partitionId, PartitionLoadInfo loadInfo)
             {
                 var blobDirectory = this.blobContainer.GetDirectoryReference($"p{partitionId:D2}");
                 var blob = blobDirectory.GetBlockBlobReference("loadinfo.json");
-                var json = JsonConvert.SerializeObject(loadInfo, Formatting.Indented);
+                var json = JsonConvert.SerializeObject(loadInfo, Formatting.Indented, serializerSettings);
                 return blob.UploadTextAsync(json, cancellationToken);
             }
 
             List<Task> tasks = info.Select(kvp => UploadPartitionInfo(kvp.Key, kvp.Value)).ToList();
-            return Task.WhenAll(tasks);           
+            return Task.WhenAll(tasks);
+        }
+
+        public async Task<T> ReadJsonBlobAsync<T>(CloudBlockBlob blob, bool throwIfNotFound, bool throwOnParseError, CancellationToken token) where T : class
+        {
+            try
+            {
+                var jsonText = await blob.DownloadTextAsync(token).ConfigureAwait(false);
+                return JsonConvert.DeserializeObject<T>(jsonText);
+            }
+            catch (StorageException e) when (!throwIfNotFound && e.RequestInformation?.HttpStatusCode == 404)
+            {
+                // container or blob does not exist
+            }
+            catch (JsonException) when (!throwOnParseError)
+            {
+                // cannot parse content of blob
+            }
+            catch(StorageException e) when (e.InnerException is OperationCanceledException operationCanceledException)
+            {
+                // unwrap the cancellation exception
+                throw operationCanceledException;
```

Comment:
I don't think I know what that means. But it probably does not matter since cancellations are not "real" exceptions, i.e. nothing went wrong and no one is going to care where it was thrown.